internal/heartbeat/heartbeat.go

```diff
@@ -173,9 +203,14 @@ func (s *Heartbeat) send(data *pb.Data) error {
 	s.reg.RegisterDevice()
 	// Sending again the heartbeat info with the right info.
 	_, err = s.dispatcherClient.Send(ctx, data)
+
 	return err
 }
 
+func isResponseEmpty(response *pb.Response) bool {
+	return len(response.Response) <= 0
```

Comment:
why the `<`?. To be honest `len(response.Response) == 0` is clear enough. IMO you don't need a function here

Reply:
good point for `nil` testing response 
And I want to keep it in a function in case the way we test for empty response changes: at first, I converted the response into a JSON string and I tested if the JSON string was `{}`, but then I found a better way. So if it become more complicated, at least, we only have to update the function and that will have no impact on the overall readability